pyiceberg/io/pyarrow.py

```diff
@@ -1441,11 +1441,15 @@ def _task_to_record_batches(
 
             # Apply the user filter
             if pyarrow_filter is not None:
-                current_batch = current_batch.filter(pyarrow_filter)
+                # Temporary fix until PyArrow 21 is released ( https://github.com/apache/arrow/pull/46057 )
+                table = pa.Table.from_batches([current_batch])
+                table = table.filter(pyarrow_filter)
                 # skip empty batches
-                if current_batch.num_rows == 0:
+                if table.num_rows == 0:
                     continue
 
+                current_batch = table.combine_chunks().to_batches()[0]
```

Comment:
I don't think we need to combine the chunks:
```suggestion
                current_batch = table.to_batches()[0]
```

Reply:
This was borrowed from the PyArrow code: https://github.com/apache/arrow/blob/main/python/pyarrow/acero.py#L379

My knowledge might be a bit limited to understand if it's needed in this case.

Reply:
Ah, I think you're right. I was confusing the batches and chunks 👍